Prusa MK3S configuration for RatOS v2.x

jfestrada opened the pull request:
I have take some work of the pull request #18 and adapted to fit on the v2.x of RatOS also I have tested all the limits, include some missing pins, and the configuration has been tested by various users. 
The speed limit has been set to 240mm/s and the microsteps for the XYZ axis has been set to 32, beyond of  260mm/s the MCU will be overloaded and klipper will restart. 

The rest of the configuration is pretty similar to the Prusa Mini.

jfestrada committed:
final configuration for Prusa Mini a Prusa Mini Bear

jfestrada committed:
remove save_config section

jfestrada committed:
Fixed control tag for the heated_bed

jfestrada committed:
Fixed PID for extruder and heated bed

jfestrada committed:
Added z_offset to the template file

jfestrada committed:
Actualizar prusa-mini.cfg

Delete the pid from the prusa-mini.cfg

jfestrada committed:
Actualizar prusa-mini-printer.template.cfg

Added the pid for the heater bed to the template

jfestrada committed:
Delete override of LOAD_FILAMENT, UNLOAD_FILAMENT, PAUSE, RESUME and CANCEL_PRINT

jfestrada committed:
Merge branch 'v2.x-PrusaMini' of github.com:ComunidadMaker3D/CM3D-klipper-configuration into v2.x-PrusaMini

jfestrada committed:
Requested changes for PR 93

jfestrada committed:
Update of the length distances for load/unload actions according with the Prusa Mini bowden tube.

jfestrada committed:
As the Buddy board does not have a clear instructions of how to connect
the ADLX345 directly to the board, the best approach is to connect it to
the RPI, this way there is no place to the error or mistakes.

input-shaper-single.cfg will have now the configuration to use the RPI
as host of the ADLX345.

jfestrada committed:
Merge branch 'v2.x-PrusaMini' into v2.x

jfestrada committed:
First mk3s+ configuration

jfestrada committed:
First mk3s+ configuration

jfestrada committed:
Fixed the offset of the Prusa Mini Z probe, this was very close to the offset of the EVA3 because of that I didn't notice before

jfestrada committed:
Fixed tach pins

jfestrada committed:
Fixed tach pins

jfestrada committed:
Trying to fix the homming

jfestrada committed:
Trying to fix the homming

jfestrada committed:
Compile scripts, macros and display for the einsy and the mk3s/mk3s+

jfestrada committed:
Compile scripts, macros and display for the einsy and the mk3s/mk3s+

jfestrada committed:
Update Z max position

jfestrada committed:
Update Z max position

jfestrada committed:
Merge branch 'v2.x' of github.com:Rat-OS/RatOS-configuration into v2.x

jfestrada committed:
Update paths in order to work with the Beta 2

jfestrada committed:
Merge branch 'v2.x-PrusaMK3S' of github.com:ComunidadMaker3D/CM3D-klipper-configuration into v2.x-PrusaMK3S

jfestrada committed:
Fix einsy board definition for display pinout

jfestrada committed:
Set microsteps to 32 and driver_SGT to 5 in order to avoid false positives

jfestrada committed:
Removed current and sense resistor from the driver configuration and moved to the motor config

jfestrada committed:
Adapted the speed limits according to the limits of the einsy MCU in order
to avoid the overload of the MCU, the speed limit performance does not make
sense if you are using the einsy board and the tmc2130 has been updated.

jfestrada committed:
Update of the heatbed's screw location

jfestrada committed:
Update the lenght of load/unload variables

jfestrada committed:
Update the reference of the tilt screw points

miklschmidt commented on templates/prusa-mk3s-printer.template.cfg:
```diff
@@ -0,0 +1,229 @@
+# STEALTH MODE
+# NOTE: You still need to include one of the above stepper motor definitions.
+# NOTE: This will make your printer quiter but less accurate, it's an inherent side effect of stealthchop.
+[include RatOS/printers/prusa-mk3s/tmc2130-stealth.cfg]
```
This does not actually run stealthchop.

jfestrada replied:
Solved, I have created a separate file in order to configure the stealthchop, and also I corrected the values

miklschmidt commented on printers/prusa-mk3s/tmc2130-stealth.cfg:
```diff
@@ -0,0 +1,78 @@
+driver_PWM_AMPL: 240
+driver_PWM_AUTOSCALE: True
+driver_SGT: 3
+stealthchop_threshold: 1
```
The driver and current settings should go in the stepper configs. Furthermore this should just be called tmc2130.cfg.

Furthermore:
1) Remove hold_current
2) Disable interpolation
3) Raise microsteps in steppers.cfg to 64 to make the steppers quieter after removing interpolation.

The above 3 changes makes the printer more accurate and less likely to cause artifacts due to runtime driver settings changes.

The extruder motor needs it's own stepper/ config too.

jfestrada replied:
Done everything execpt the microsteps change because the einsy is not able to work at 64 microsteps

miklschmidt replied:
@jfestrada aah that's fair enough, forgot we are dealing with a 16mhz 8-bit controller here. Hmm. I kind of want to isolate those settings to the einsy boards so it's not hampering people if they decide to install a modern 32-bit board. Let me think about that!

jfestrada replied:
I was thinking about this also, and to my mind the best approach would be to rename the file so we could include the file steppers-einsy-board.cfg and steppers.cfg, the default should be the steppers-einsy-board.cfg. Take into account also that everything has been configured to work with the tmc2130. Also another thing to have in mind is that this is a "low cost" upgrade for the people that have a mk3s and want a little bit more, so they don't need to change the board at least that the board die.

miklschmidt replied:
@jfestrada You're right about that last part. Since most of the steppers section kind of falls into "advanced usage", i guess it's OK for now to have the steppers.cfg use 32 microsteps. Let's leave it as it is 👍

miklschmidt commented on templates/prusa-mk3s-printer.template.cfg:
```diff
@@ -0,0 +1,229 @@
+position_max: 214
+
+[probe]
+z_offset = 2.0
```
This should be 0 to avoid accidents.

jfestrada replied:
Yes, you are right I noticed it in my installation

miklschmidt commented on templates/prusa-mk3s-printer.template.cfg:
```diff
@@ -0,0 +1,229 @@
+#############################################################################################################
+
+[mcu]
+serial: /dev/prusa-einsy
```
Delete this, this should be in the board config.

jfestrada replied:
Done

miklschmidt commented on printers/prusa-mk3s/steppers.cfg:
```diff
@@ -0,0 +1,33 @@
+step_pin: x_step_pin
+dir_pin: !x_dir_pin
+enable_pin: !x_enable_pin
+microsteps: 32
```
Change all microstep settings to 64 for noise reduction when interpolation is off.

jfestrada replied:
There is no way to use it with 64 microsteps, maximum is 32

![image](https://user-images.githubusercontent.com/7290345/221525362-ac59f866-69d8-4175-a724-5b0fb98e6917.png)

miklschmidt commented on printers/initial-setup.cfg:
```diff
@@ -59,6 +59,16 @@ command: /home/pi/printer_data/config/RatOS/scripts/install-prusa-mini-config.sh
 verbose: False
 
+[gcode_shell_command copy_prusa_mk3s]
+command: /home/pi/klipper_config/config/scripts/install-prusa-mk3s-config.sh
```
change `klipper_config` to `printer_data/config`

jfestrada replied:
Done

miklschmidt commented on printers/initial-setup.cfg:
```diff
@@ -59,6 +59,16 @@ command: /home/pi/printer_data/config/RatOS/scripts/install-prusa-mini-config.sh
+verbose: False
+
+[gcode_shell_command copy_prusa_mk3s]
+command: /home/pi/klipper_config/config/scripts/install-prusa-mk3s-config.sh
```
change `klipper_config` to `printer_data/config`

jfestrada replied:
Done

miklschmidt commented on scripts/compile-binaries.sh:
```diff
@@ -162,6 +162,15 @@ compile_prusa_buddy() {
+    make olddefconfig
+    make clean
+    make
+    cp /home/pi/klipper/out/klipper.bin /home/pi/printer_data/firmware_binaries/firmware-prusa-einsy.bin
```
change `printer_data` to `printer_data/config`

jfestrada replied:
Done

miklschmidt commented on scripts/compile-binaries.sh:
```diff
@@ -162,6 +162,15 @@ compile_prusa_buddy() {
     cp /home/pi/klipper/out/klipper.bin /home/pi/printer_data/firmware_binaries/firmware-prusa-buddy.bin
```
Just cought this one from the earlier PR, change `printer_data` to `printer_data/config`

jfestrada replied:
Done

miklschmidt reviewed (changes requested):
Thank you! just a couple of things to fix, and we should be good to go :)

jfestrada committed:
Fixed drivers and steppers configurations, also some paths have been corrected

miklschmidt commented on boards/prusa-einsy/config.cfg:
```diff
@@ -36,5 +36,5 @@ aliases:
 
 [mcu]
-serial: /dev/prusa-i3-mk3sp
+serial: /dev/ttyACM0
```
/dev/prusa-i3-mk3sp doesn't work? Assuming it's flashed with firmware compiled with the config from RatOS, and you've run ratos-update.sh to symlink the udev rule, it should work.

jfestrada replied:
There was a mistake with the name of the device, in the rules file was called /dev/prusa-einsy, so finally I left that way.

miklschmidt replied:
Looks good now 👍

miklschmidt commented on printers/prusa-mk3s/tmc2130.cfg:
```diff
@@ -0,0 +1,70 @@
+driver_PWM_GRAD: 4
+driver_PWM_AMPL: 240
+driver_PWM_AUTOSCALE: True
+driver_SGT: 3
```
driver_* settings still need to go into the appropriate steppers/* file, so steppers/ldo/42sth40-1684/2130/24v-0.4a-x.cfg for example, since they're specific to the driver/stepper combination.

jfestrada replied:
Done

miklschmidt commented on steppers/ldo/42sth40-1684ac/2130/24v-0.4a-x.cfg:
```diff
@@ -1,6 +1,7 @@
 [tmc2130 stepper_x]
 sense_resistor: 0.22
 run_current: 0.4
+hold_current: 0.2
```
Delete this (see: https://www.klipper3d.org/TMC_Drivers.html#prefer-to-not-specify-a-hold_current)

jfestrada replied:
I noticed that two days ago, and was in my TODO, so I tested and is working fine (as expected)

miklschmidt commented on steppers/ldo/42sth40-1684ac/2130/24v-0.4a-y.cfg:
```diff
@@ -1,6 +1,7 @@
 [tmc2130 stepper_y]
 sense_resistor: 0.22
 run_current: 0.4
+hold_current: 0.2
```
Delete this (see: https://www.klipper3d.org/TMC_Drivers.html#prefer-to-not-specify-a-hold_current)

jfestrada replied:
Done

jfestrada replied:
Done

miklschmidt commented on steppers/ldo/42sth40-1684ac/2130/24v-0.4a-z.cfg:
```diff
@@ -1,6 +1,7 @@
 [tmc2130 stepper_z]
 sense_resistor: 0.30
 run_current: 0.4
+hold_current: 0.2
```
Delete this (see: https://www.klipper3d.org/TMC_Drivers.html#prefer-to-not-specify-a-hold_current)

jfestrada replied:
Done

miklschmidt commented on steppers/ldo/42sth40-1684ac/2130/24v-0.4a-extruder.cfg:
```diff
@@ -0,0 +1,8 @@
+[tmc2130 extruder]
+sense_resistor: 0.22
+run_current: 0.52
+hold_current: 0.2
```
Delete this (see: https://www.klipper3d.org/TMC_Drivers.html#prefer-to-not-specify-a-hold_current)

jfestrada replied:
Done

miklschmidt commented on templates/prusa-mk3s-printer.template.cfg:
```diff
@@ -212,10 +201,14 @@ recover_velocity: 50.
 
 [probe]
-z_offset = 2.0
+z_offset = -0.0
```
remove  `-`  so it's just `0.0`

jfestrada replied:
Done, I use "-" in order to indicate that should be a negative value

jfestrada committed:
Fixed udev link, remove the hold_current as specified by good
practice of klipper and moved all the driver* configuration to
the appropiate stepper file.

jfestrada committed:
z_offset default value corrected, now is 0.0 instead -0.0

miklschmidt commented:
Thank you for your work @jfestrada!
